include/seqan3/alphabet/translation/translation_details.hpp

```diff
@@ -0,0 +1,466 @@
+
+#include <seqan3/alphabet/nucleotide/all.hpp>
+#include <seqan3/alphabet/aminoacid/all.hpp>
+
+namespace seqan3
+{
+/*!\brief Specialization values for genetic codes.
+ * \details The numeric values of the enums correspond to the genbank transl_table values
+ * (see http://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi).
+ */
+enum struct genetic_code : uint8_t
+{
+    CANONICAL=1,
+    VERT_MITOCHONDRIAL,
+    YEAST_MITOCHONDRIAL,
+    MOLD_MITOCHONDRIAL,
+    INVERT_MITOCHONDRIAL,
+    CILIATE,
+    FLATWORM_MITOCHONDRIAL = 9,
+    EUPLOTID,
+    PROKARYOTE,
+    ALT_YEAST,
+    ASCIDIAN_MITOCHONDRIAL,
+    ALT_FLATWORM_MITOCHONDRIAL,
+    BLEPHARISMA,
+    CHLOROPHYCEAN_MITOCHONDRIAL,
+    TREMATODE_MITOCHONDRIAL = 21,
+    SCENEDESMUS_MITOCHONDRIAL,
+    THRAUSTOCHYTRIUM_MITOCHONDRIAL,
+    PTEROBRANCHIA_MITOCHONDRIAL,
+    GRACILIBACTERIA
+};
+
+template <typename nucl_type, genetic_code, typename void_type = void>
+struct translation_table
+{
+    static aa27 const VALUE[nucl_type::value_size][nucl_type::value_size][nucl_type::value_size];
+};
+
+template <typename nucl_type, typename void_type>
+struct translation_table<nucl_type, genetic_code::CANONICAL, void_type>
+{
+    static aa27 const VALUE[nucl_type::value_size][nucl_type::value_size][nucl_type::value_size];
+};
+
+template <typename void_type>
+struct translation_table<dna4, genetic_code::CANONICAL, void_type>
+{
+    static aa27 const VALUE[dna4::value_size][dna4::value_size][dna4::value_size];
+};
+
+//!\brief Translation table for canonical genetic code and dna4 alphabet.
+template <typename void_type>
+aa27 const translation_table<dna4, genetic_code::CANONICAL, void_type>::VALUE[dna4::value_size][dna4::value_size][dna4::value_size] =
+{
+    { // a??
+        { aa27::K,          aa27::N, aa27::K,          aa27::N }, // aa?
+        { aa27::T,          aa27::T, aa27::T,          aa27::T }, // ac?
+        { aa27::R,          aa27::S, aa27::R,          aa27::S }, // ag?
+        { aa27::I,          aa27::I, aa27::M,          aa27::I }  // au?
+    }, { // c??
+        { aa27::Q,          aa27::H, aa27::Q,          aa27::H }, // ca?
+        { aa27::P,          aa27::P, aa27::P,          aa27::P }, // cc?
+        { aa27::R,          aa27::R, aa27::R,          aa27::R }, // cg?
+        { aa27::L,          aa27::L, aa27::L,          aa27::L }  // cu?
+    }, { // g??
+        { aa27::E,          aa27::D, aa27::E,          aa27::D }, // ga?
+        { aa27::A,          aa27::A, aa27::A,          aa27::A }, // gc?
+        { aa27::G,          aa27::G, aa27::G,          aa27::G }, // gg?
+        { aa27::V,          aa27::V, aa27::V,          aa27::V }  // gu?
+    }, { // u??
+        { aa27::TERMINATOR, aa27::Y, aa27::TERMINATOR, aa27::Y }, // ua?
+        { aa27::S,          aa27::S, aa27::S,          aa27::S }, // uc?
+        { aa27::TERMINATOR, aa27::C, aa27::W,          aa27::C }, // ug?
+        { aa27::L,          aa27::F, aa27::L,          aa27::F }  // uu?
+    }
+};
```

Comment:
You can generate the tables of the smaller nucleotide alphabets, by accessing the values in the nucl16's table via a constexpr lambda. Similar to:
https://github.com/seqan/seqan3/blob/master/include/seqan3/alphabet/nucleotide/dna4.hpp#L298

Actually, you could make it the default case for any nucleotide alphabet and then just make a single template specialisation for nucl16!